wgpu-core/src/device/resource.rs

```diff
@@ -3317,7 +3321,8 @@ impl<A: HalApi> Device<A> {
         self.valid.store(false, Ordering::Release);
 
         // 1) Resolve the GPUDevice device.lost promise.
-        let closure = self.lock_life().device_lost_closure.take();
+        let mut life_lock = self.lock_life();
```

Comment:
We shouldn't be holding the life lock across external closure invocations - this could potentially deadlock.